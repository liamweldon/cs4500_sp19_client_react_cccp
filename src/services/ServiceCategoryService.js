export default class ServiceCategoryService {
    static instance = null;
    static getInstance() {
        if(ServiceCategoryService.instance === null) {
            ServiceCategoryService.instance = new ServiceCategoryService()
        }
        return this.instance
    }
    findServiceCategoryById = categoryId =>
        fetch(`http://localhost:8080/api/service-categories/${categoryId}`)
            .then(response => response.json())
    findAllServiceCategories = () =>
        fetch("http://localhost:8080/api/service-categories")
            .then(response => response.json())
}
