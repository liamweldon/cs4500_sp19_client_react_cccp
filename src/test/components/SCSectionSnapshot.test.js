import React from "react";
import TestRenderer from "react-test-renderer";

import ServiceCategorySection from "../../components/ServiceNavigator/ServiceCategorySection";
import ServiceCategoryService from "../../services/ServiceCategoryService";
import mockService from "../services/mocks/MockServiceCategoryService";
import serviceCategories from '../mockData/service-categories.mock.json'

const serviceCategoryService = ServiceCategoryService.getInstance();
beforeEach(() => {
  global.fetch = mockService(serviceCategories);
});

// Provide a new service to insert
const test_new_service = {
  id: 555,
  serviceName: "Car Wash",
  serviceDescription: "Clean your car!"
};


// Snapshot testing code
test("Rerender works properly", () => {
  // Create snapshot of initial services
  return serviceCategoryService
    .findServiceCategoryById(123)
    .then(category => {
      let component = TestRenderer.create(
        <ServiceCategorySection serviceCategory={category} />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      // Add a new service to the category's service list, and re-render. Since there's 
      // No real interactivity changing the dom in this component, this is the only
      // real way to test rendering changes.
      category.services.push(test_new_service);

      component = TestRenderer.create(
            <ServiceCategorySection serviceCategory={category} />);
      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    })
    .catch(function(error) {});
});