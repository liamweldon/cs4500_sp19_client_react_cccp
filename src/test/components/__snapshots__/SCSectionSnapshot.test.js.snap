// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rerender works properly 1`] = `
<div>
  <a
    id="123"
  />
  <h2>
    Home Improvements
  </h2>
  <div>
    <div
      className="row"
    >
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Appliance Installation
            </a>
          </p>
        </div>
      </div>
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Cabinetry
            </a>
          </p>
        </div>
      </div>
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Door Installation
            </a>
          </p>
        </div>
      </div>
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Concrete Removal
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Appliance Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Cabinetry
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Door Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Concrete Removal
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Dishwasher Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Finish Carpentry
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Floor Polishing
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Furniture Assembly
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Generator Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Gardening
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Home Organizing
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Electrical and Wiring Repair
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Rerender works properly 2`] = `
<div>
  <a
    id="123"
  />
  <h2>
    Home Improvements
  </h2>
  <div>
    <div
      className="row"
    >
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Appliance Installation
            </a>
          </p>
        </div>
      </div>
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Cabinetry
            </a>
          </p>
        </div>
      </div>
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Door Installation
            </a>
          </p>
        </div>
      </div>
      <div
        className="card no-border col-6"
      >
        <img
          className="card-img-top"
        />
        <div
          className="card-body"
        >
          <p
            className="card-text"
          >
            <a
              href="/provider"
            >
               
              Concrete Removal
            </a>
          </p>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Appliance Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Cabinetry
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Door Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Concrete Removal
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Dishwasher Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Finish Carpentry
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Floor Polishing
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Furniture Assembly
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Generator Installation
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Gardening
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Home Organizing
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Electrical and Wiring Repair
        </a>
      </div>
      <div
        className="col-6 list-group-item no-border"
      >
        <a
          href="/provider"
        >
           
          Car Wash
        </a>
      </div>
    </div>
  </div>
</div>
`;
